src/components/Modal.astro

```diff
@@ -0,0 +1,91 @@
+---
+interface Props {
+  modalContent: {
+    title?: string;
+    description?: string;
+  };
+}
+
+const {
+  modalContent: { title, description },
+} = Astro.props;
+---
+
+<div class="fixed hidden inset-0" data-visible="false" id="modal-wrapper">
+  <div class="absolute inset-0 backdrop-blur-sm z-10"></div>
+  <div class="absolute inset-0 z-20 flex justify-center items-center">
+    <div class="max-w-2xl w-full">
+      <div
+        class="modal-container mx-8 p-6 relative flex flex-col bg-sky-200 border-sky-400 border-2 md:p-10"
+      >
+        <div class="flex justify-center">
+          <button class="btn-close p-2 absolute w-12 h-12 top-0 right-0">
+            <img src=`${import.meta.env.BASE_URL}/cross.svg` alt="cross icon" />
+            <span class="sr-only">Cross icon to close modal</span>
+          </button>
+          <slot name="title">
+            <h2 class="text-2xl md:text-4xl">{title}</h2>
+          </slot>
+        </div>
+        <div>
+          <slot>
+            <p class="mt-8 text-base overflow-auto max-h-[40vh] md:text-xl">
+              {description}
+            </p>
+          </slot>
+          <div class="mt-10 flex justify-between w-full">
+            <button class="btn btn-close" type="button">close</button>
+            <button class="btn" type="submit">confirm</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+  function modal() {
+    const wrapper = document.querySelector<HTMLElement>("#modal-wrapper")!;
+    const container = document.querySelector<HTMLElement>(".modal-container")!;
+    const triggerBtn = document.querySelector<HTMLButtonElement>(".btn-offer")!;
+    const closeModalBtns =
+      wrapper.querySelectorAll<HTMLButtonElement>(".btn-close");
+
+    function close() {
+      wrapper.dataset.visible = "false";
+      document.removeEventListener("click", closeOutside, true);
+    }
+
+    function closeOutside(e) {
+      if (!container.contains(e.target)) {
+        close();
+      }
+    }
+
+    function open() {
+      wrapper.dataset.visible = "true";
+      document.addEventListener("click", closeOutside, true);
+    }
+
+    triggerBtn.addEventListener("click", open);
+
+    closeModalBtns.forEach((closeModal) =>
+      closeModal.addEventListener("click", close)
+    );
+  }
+  document.addEventListener("DOMContentLoaded", modal);
+</script>
+
+<style>
```

Comment:
We could improve this a bit
```suggestion
<style scoped>
```
https://docs.astro.build/en/guides/styling/#scoped-styles